test/test_01_guess_cmake.py

```diff
@@ -0,0 +1,33 @@
+
+import time
+import pytest
+import config
+import subprocess
+
+def ExecutablesOfBuffer(path):
+    return "guess_executable_cmake#ExecutablesOfBuffer('" + path + "')"
+
+def test_cmake_completion(eng):
+    subprocess.run(["cmake", "src", "-B", "src/build"])
+    eng.exe("cd src")
```

Comment:
This may be the reason for the consequent test failures. Please make sure to restore the previous working directory. Probably, a ptest fixture could be best suitable for this.

Reply:
I tried using a pytest fixture, but I couldn't get it to work. It kept executing *after* the test instead of before it. I tried running `bd` and `cd ../` after the test completes and I still get test errors on my computer. I'm not sure what to do here.

Reply:
There're good examples of pytest fixtures in [conftest.py](https://github.com/sakhnik/nvim-gdb/blob/master/test/conftest.py). Basically, a fixture is just a decorated function like:
```python
@pytest.fixture...
def cd_to_cmake(eng):
    eng.exe("cd src")
    yield
    eng.exe("cd ..")
```
It's set up automatically as soon as passed into a test method, just like `eng`:
```python
def test_cmake_completion(eng, cd_to_cmake):
    # Something to check after the directory has been changed
```
I'm going to try the tests on my machine to see what's going on.

Reply:
I was confused about the `yield`. I've got it now (though many of the tests still fail on my computer)